### PR TITLE
Save the last chosen menu entry into config file

### PR DESCRIPTION
**Pull request by gitmpr**
Save the last chosen menu entry into config file, read the index upon loading the menu and make it the selected one

### CODE REVIEW

**Comment by mmeyer724** [comment]
Hi there! It's been a moment since I last worked on this project, but I'm happy to see contributions still rolling through 🙂 

This is a great addition, but one thing I would suggest is splitting the config reading/writing out into separate functions, e.g:
```
def get_last_selected_target_index():
    config = json.loads(resources.user.read(config_name))
    return config.get('saved_last_chosen_index', 0)


def save_last_selected_target_index(index: int):
    config = json.loads(resources.user.read(config_name))
    config['saved_last_chosen_index'] = index
    resources.user.write(config_name, json.dumps(config, indent=4))
```
We can then call these functions from `display_menu`, in the same places where you are reading and writing the config currently. This change will help in keeping the size of the `display_menu` function down, as it's already quite complex. 

Please let me know what you think! Thank you 🙏🏻

**Comment by gitmpr** [comment]
That's a good way to incorporate it in a more readable manageable way. Nice that you consider it so quickly even though the project is somewhat old. I still got a great use out of it. I considered creating something similar with the [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu) library but didn't want to have to write something that probably already existed. I still might though as being able to search through all the entries in the menu would be nice

File: sshmenu/sshmenu.py [hunk 1]
```diff
@@ -215,14 +216,25 @@ def update_targets():
...
+
+def save_last_selected_target_index(index: int):
+    config = json.loads(resources.user.read(config_name))
+    config['saved_last_chosen_index'] = index
+    resources.user.write(config_name, json.dumps(config, indent=4))
+
+
 def display_menu():
-    global targets
+    global targets, config_name
```
Comment (mmeyer724) [inline sshmenu/sshmenu.py:231]: minor nit: we don't need the `config_name` variable here anymore 

(side note: I'm pretty sure we can remove this entire line, as `global` is only necessary if you want to modify global variables from a local context, which we are not doing. But I'm okay with leaving it in place anyways)

File: sshmenu/sshmenu.py [hunk 2]
```diff
@@ -393,3 +408,4 @@ def hook():
     result = input(prompt)
     readline.set_pre_input_hook()
     return result
+
```
Comment (mmeyer724) [inline sshmenu/sshmenu.py:411]: minor nit: empty line

**Review by mmeyer724 — APPROVED** [review]

**Comment by mmeyer724** [comment]
> That's a good way to incorporate it in a more readable manageable way. Nice that you consider it so quickly even though the project is somewhat old. I still got a great use out of it. I considered creating something similar with the [simple-term-menu](https://github.com/IngoMeyer441/simple-term-menu) library but didn't want to have to write something that probably already existed. I still might though as being able to search through all the entries in the menu would be nice

I left two minor nit comments but otherwise this lgtm! Thank you 

Btw, if you want searching you should take a look at [fzf](https://github.com/junegunn/fzf). You might be able to do something cool with it

**Comment by gitmpr** [comment]
Alright, great. How to proceed now? Do you want me to make another pull request with the changes you mentioned so that you can merge it with the master branch? Is it not possible on Github to edit my pull request and then merge it?

**Comment by mmeyer724** [comment]
> Alright, great. How to proceed now? Do you want me to make another pull request with the changes you mentioned so that you can merge it with the master branch? Is it not possible on Github to edit my pull request and then merge it?

I'm actually going to merge it now as-is (given my comments were just nits anyways), and I'll resolve the two small nits in a follow up alongside a version bump 

Thank you for the contribution!

**Pull request merged**